src/SearchResults/generate-hierarchical-tree.js

```diff
@@ -59,7 +61,10 @@ function generateHierarchicalTree(sortBy, hierarchicalSeparator, currentRefineme
       //
       // If parent refinement is `beers`, then we do not want to have `bières > Belges`
       // showing up
-      var onlyMatchingValuesFn = filterFacetValues(parent.path, currentRefinement, hierarchicalSeparator);
+
+      var onlyMatchingValuesFn = filterFacetValues(parent.path || hierarchicalPrefixPath,
```

Comment:
This test fails is parent.path is an empty string, or '0'. Can you compute it with the right default values/undefined?